src/theme/components/image.ts

```diff
@@ -1,17 +1,32 @@
 interface CustomImageBaseStyle {
-  size: keyof typeof imageSizes;
+  ratio: keyof typeof imageRatios;
+  size: keyof typeof imageSizes | keyof typeof imageSizesBasedOnHeight;
+  sizeBasedOn: "height" | "width";
 }
 interface CustomImageWrapperBaseStyle {
```

Comment:
It looks like both interfaces have the same structure, so you can probably rename it and delete the other.